src/main/scala/strawman/collection/immutable/Array.scala

```diff
@@ -0,0 +1,84 @@
+package strawman.collection.immutable
+
+import strawman.collection.mutable.{ArrayBuffer, ArrayBufferView, Iterator}
+import strawman.collection.{Iterable, IterableFactory, IterableOnce, IndexedSeq, Seq, SeqLike}
+
+import scala.{AnyRef, Boolean, Int}
+import scala.Predef.{???, intWrapper}
+
+/**
+  * An immutable array.
+  *
+  * Supports efficient indexed access and has a small memory footprint.
+  */
+class Array[+A] private (private val elements: scala.Array[AnyRef]) extends IndexedSeq[A] with SeqLike[A, Array] {
```

Comment:
There is a companion’s `apply` method (inherited from `IterableFactory`) which creates an `ArrayBuffer` and then copies its contents into a fresh new `Array`.

You think we should also add a public constructor aliasing an `Array` passed as parameter?